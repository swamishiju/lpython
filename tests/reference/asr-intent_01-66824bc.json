{
    "basename": "asr-intent_01-66824bc",
    "cmd": "lpython --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/intent_01.py",
    "infile_hash": "0a8fa2940567fccee2cfccd4af40f353b74dbe542590460a195246a5",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-intent_01-66824bc.stdout",
    "stdout_hash": "2ead5794da4c917371e75c62e1d44e27092c8790d76efc1916b7aee8",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}