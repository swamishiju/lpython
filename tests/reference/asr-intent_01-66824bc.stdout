(TranslationUnit
    (SymbolTable
        1
        {
            __main__:
                (Module
                    (SymbolTable
                        2
                        {
                            Foo:
                                (Struct
                                    (SymbolTable
                                        3
                                        {
                                            p:
                                                (Variable
                                                    3
                                                    p
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Integer 4)
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                    .false.
                                                    .false.
                                                    ()
                                                    .false.
                                                    .false.
                                                )
                                        })
                                    Foo
                                    (StructType
                                        [(Integer 4)]
                                        []
                                        .true.
                                        .false.
                                    )
                                    []
                                    [p]
                                    []
                                    Source
                                    Public
                                    .false.
                                    .false.
                                    [(())]
                                    ()
                                    ()
                                ),
                            f:
                                (Function
                                    (SymbolTable
                                        4
                                        {
                                            w:
                                                (Variable
                                                    4
                                                    w
                                                    []
                                                    Out
                                                    ()
                                                    ()
                                                    Default
                                                    (Array
                                                        (StructType
                                                            [(Integer 4)]
                                                            []
                                                            .true.
                                                            .false.
                                                        )
                                                        [((IntegerConstant 0 (Integer 4) Decimal)
                                                        (IntegerConstant 5 (Integer 4) Decimal))]
                                                        PointerArray
                                                    )
                                                    2 Foo
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                    .false.
                                                    .false.
                                                    ()
                                                    .false.
                                                    .false.
                                                ),
                                            x:
                                                (Variable
                                                    4
                                                    x
                                                    []
                                                    In
                                                    ()
                                                    ()
                                                    Default
                                                    (Integer 4)
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                    .false.
                                                    .false.
                                                    ()
                                                    .false.
                                                    .false.
                                                ),
                                            y:
                                                (Variable
                                                    4
                                                    y
                                                    []
                                                    In
                                                    ()
                                                    ()
                                                    Default
                                                    (Real 8)
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                    .false.
                                                    .false.
                                                    ()
                                                    .false.
                                                    .false.
                                                ),
                                            z:
                                                (Variable
                                                    4
                                                    z
                                                    []
                                                    InOut
                                                    ()
                                                    ()
                                                    Default
                                                    (List
                                                        (UnsignedInteger
                                                            4
                                                        )
                                                    )
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                    .false.
                                                    .false.
                                                    ()
                                                    .false.
                                                    .false.
                                                )
                                        })
                                    f
                                    (FunctionType
                                        [(Integer 4)
                                        (Real 8)
                                        (List
                                            (UnsignedInteger
                                                4
                                            )
                                        )
                                        (Array
                                            (StructType
                                                [(Integer 4)]
                                                []
                                                .true.
                                                .false.
                                            )
                                            [((IntegerConstant 0 (Integer 4) Decimal)
                                            (IntegerConstant 5 (Integer 4) Decimal))]
                                            PointerArray
                                        )]
                                        ()
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        .false.
                                    )
                                    []
                                    [(Var 4 x)
                                    (Var 4 y)
                                    (Var 4 z)
                                    (Var 4 w)]
                                    []
                                    ()
                                    Public
                                    .false.
                                    .false.
                                    ()
                                )
                        })
                    __main__
                    ()
                    []
                    .false.
                    .false.
                    .false.
                ),
            main_program:
                (Program
                    (SymbolTable
                        6
                        {
                            
                        })
                    main_program
                    []
                    []
                )
        })
    []
)
